scipy/stats/_resampling.py

```diff
@@ -1401,34 +1396,34 @@ def _batch_generator(iterable, batch):
 
 
 def _pairings_permutations_gen(n_permutations, n_samples, n_obs_sample, batch,
-                               random_state):
+                               rng):
     # Returns a generator that yields arrays of size
     # `(batch, n_samples, n_obs_sample)`.
     # Each row is an independent permutation of indices 0 to `n_obs_sample`.
     batch = min(batch, n_permutations)
 
-    if hasattr(random_state, 'permuted'):
+    if hasattr(rng, 'permuted'):
         def batched_perm_generator():
             indices = np.arange(n_obs_sample)
             indices = np.tile(indices, (batch, n_samples, 1))
             for k in range(0, n_permutations, batch):
                 batch_actual = min(batch, n_permutations-k)
                 # Don't permute in place, otherwise results depend on `batch`
-                permuted_indices = random_state.permuted(indices, axis=-1)
+                permuted_indices = rng.permuted(indices, axis=-1)
```

Comment:
Maybe? Note that this section is doing different things with `RandomState`s and `Generator`s for a reason. Half of this can go away when we remove support for `RandomState`. If there is additional simplification possible for `Generator`s, it seems separate from this PR, right?

Reply:
Both `Generator` and `RandomState` have the `permutation` method. I think the method for each type shuffles slices independently, and if `indices` is an array the shuffling is done on a copy.

Also, I think `Generator.permute(indices, axis=-1)` and `Generator.permutation(indices, axis=-1)` do the same thing. It's not clear to me why there are two Generator methods doing the same thing.

If `RandomState.permutation` is slow then that's probably justification for sticking with status quo.

Reply:
Now I look into it further, `RandomState.permutation` only shuffles along the first axis.

Reply:
> Both `Generator` and `RandomState` have the `permutation` method. I think the method for each type shuffles slices independently, and if indices is an array the shuffling is done on a copy.

Rather than just thinking about it, let's refer to the documentation.

![image](https://github.com/user-attachments/assets/38307b7b-7f4f-49d2-8b82-8f1c52fe9e50)

```python3
import numpy as np
A = np.arange(5)
B = np.stack([A, A, A, A, A])
np.random.RandomState().permutation(B.T)
# array([[0, 0, 0, 0, 0],
#        [1, 1, 1, 1, 1],
#        [3, 3, 3, 3, 3],
#        [4, 4, 4, 4, 4],
#        [2, 2, 2, 2, 2]])
```

So no, `RandomState.permutation` does not shuffle slices independently.

> If `RandomState.permutation` is slow then that's probably justification for sticking with status quo.

What is the "status quo" in this context? If you mean leaving this code alone and keeping this scope of this PR to the transition from `random_state` to `rng`, I agree!

You can see in the blame that this code comes from gh-17030, which improved its performance by a factor of 25~50x of what it was before.